src/new_hope/week1.clj

```diff
@@ -65,3 +65,85 @@
     (crit/quick-bench (second (reverse coll)))
     (crit/quick-bench (last (butlast coll))))
   )
+
+;; Homework
+
+;; Sum It All Up
+(defn sum [coll] (if (empty? coll) 0 (+ (first coll) (sum (rest coll)))))
+
+(facts "about sum"
+       (sum [1 2 3]) => 6
+       (sum (list 0 -2 5 5)) => 8
+       (sum #{4 2 1}) => 7
+       (sum '(0 0 -1)) => -1
+       (sum '(1 10 3)) => 14)
+
+;; Find the odd numbers
+(defn odd-nums [coll] (filter odd? coll))
+
+(facts "about odd-nums"
+       (odd-nums #{1 2 3 4 5}) => '(1 3 5)
+       (odd-nums [4 2 1 6]) => '(1)
+       (odd-nums [2 2 4 6]) => '()
+       (odd-nums [1 1 1 3]) => '(1 1 1 3))
+
+;; Palindrome Detector
+(defn palindrom? [x] (= (seq x) (reverse (seq x))))
+
+(facts "about palindrom?"
+       (palindrom? '(1 2 3 4 5)) => false
+       (palindrom? "racecar") => true
+       (palindrom? [:foo :bar :foo]) => true
+       (palindrom? '(1 1 3 3 1 1)) => true
+       (palindrom? '(:a :b :c)) => false)
+
+;; Duplicate a Sequence
+(defn dup-seq [coll] (if (empty? coll) '() (conj (dup-seq (rest coll)) (first coll) (first coll))))
+
+(facts "about dup-seq"
+       (dup-seq [1 2 3]) => '(1 1 2 2 3 3)
+       (dup-seq [:a :a :b :b]) => '(:a :a :a :a :b :b :b :b)
+       (dup-seq [[1 2] [3 4]]) => '([1 2] [1 2] [3 4] [3 4])
+       (dup-seq [[1 2] [3 4]]) => '([1 2] [1 2] [3 4] [3 4]))
+
+;; Day 5
+
+;; Compress a Sequence
+(defn compress-seq [coll] (map #(first %) (partition-by #(do %) coll)))
+
+(facts "about compress-seq"
+       (apply str (compress-seq "Leeeeeerrroyyy")) => "Leroy"
+       (compress-seq [1 1 2 3 3 2 2 3]) => '(1 2 3 2 3)
+       (compress-seq [[1 2] [1 2] [3 4] [1 2]]) => '([1 2] [3 4] [1 2]))
+
+;; Pack a Sequence
+(defn pack-seq [coll] (partition-by #(do %) coll))
+
+(facts "about pack-seq"
+       (pack-seq [1 1 2 1 1 1 3 3]) => '((1 1) (2) (1 1 1) (3 3))
+       (pack-seq [:a :a :b :b :c]) => '((:a :a) (:b :b) (:c))
+       (pack-seq [[1 2] [1 2] [3 4]]) => '(([1 2] [1 2]) ([3 4])))
+
+;; Drop Every Nth Item
+(defn drop-nth [coll n] (flatten (map #(if (= (count %) n) (butlast %) %) (partition-all n coll))))
+
+(facts "about drop-nth"
+       (drop-nth [1 2 3 4 5 6 7 8] 3) => [1 2 4 5 7 8]
+       (drop-nth [:a :b :c :d :e :f] 2) [:a :c :e]
+       (drop-nth [1 2 3 4 5 6] 4) [1 2 3 5 6])
+
+;; Intro to Iterate
+(comment
+  (take 5 (iterate inc 1))
+  (= '(1 4 7 10 13) (take 5 (iterate #(+ 3 %) 1)))
+  )
+
+;; Replicate a Sequence
+(defn repl-seq [coll n] (flatten (map #(take n (repeat %)) coll)))
```

Comment:
`flatten` is dangerous, it flattens everything down to the bottom, but you need only one level. Consider using `mapcat`
Example test:
```
(repl-seq [[1 1] [2 2]] 2)
```